### ISSUE TITLE
Deprecation warning of the punycode module

### ISSUE THREAD

**Issue by ericfortis** [nodemailer/nodemailer#1587]
[Node 21.1 deprecated](https://github.com/nodejs/node/pull/47202) `punycode`

```
[DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at Object.<anonymous> (…/node_modules/nodemailer/lib/mime-node/index.js:7:18)
```

The [new documentation](https://nodejs.org/api/punycode.html#punycode) suggests:

> Users currently depending on the punycode module should switch to using the userland-provided [Punycode.js](https://github.com/bestiejs/punycode.js) module instead. For punycode-based URL encoding, see [url.domainToASCII](https://nodejs.org/api/url.html#urldomaintoasciidomain) or, more generally, the [WHATWG URL API](https://nodejs.org/api/url.html#the-whatwg-url-api).

**Comment by charbelnicolas** [comment]
Any news on this?

**Comment by andris9** [comment]
No plans to change anything for now. Nodemailer, for various reasons, has no userland dependencies, and it is fully self-contained. Creating a punycode encoder for Nodemailer so that it would work with these very old Node versions is not a priority for now. At least not until Node 22 is released.

**Comment by axkibe** [comment]
If you don't want the dependency just copy punycode.js into your source base. (License MIT)

I guess the node team made it clear, they are not maintaining it anymore, so no loss the from coping the .js instead of depending on it.. They just  left it for now for backward compatibility. 

PS: At least already now the warning message is annoying.

**Comment by AlessioGr** [comment]
> This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days.

Commenting so this issue isn't closed.

**Comment by SatyanshMittal** [comment]
solution that worked for me:
npm install punycode --save

and then in node_modules go to the directory tr46 > index.js

// Replace this:
const punycode = require('punycode');
// With this:
const punycode = require('punycode/');

basically just add a trailing forward slash

**Comment by sanchyanchakraborty** [comment]
Observe the identical solution using a custom automated script. #[1599](https://github.com/nodemailer/nodemailer/issues/1599)
@SatyanshMittal

**Comment by hulkish** [comment]
It's going to be removed from node core at some point, so probably best to actually correct this issue by depending on punycode.js - as advised from node core for this

**Comment by andris9** [comment]
Fixed in latest version

**Comment by charbelnicolas** [comment]
> Fixed in latest version

I can still see the warning with version 6.9.10, is there anything else I need to do to get rid of the warning?

![ss_2024_02_22_17_44_09](https://github.com/nodemailer/nodemailer/assets/14960414/596a990e-3760-49fe-9711-cd5340ecfeb1)

**Comment by lukasguldstoev** [comment]
Same issue. Fixed after updating to v20.11.1, deleting package-lock.json, node_modules and reinstalling

`nvm install 20.11.1
sudo rm -rf node_modules package-lock.json
npm i`

**Comment by charbelnicolas** [comment]
> Same issue. Fixed after updating to v20.11.1, deleting package-lock.json, node_modules and reinstalling
> 
> ```
> nvm install 20.11.1
> sudo rm -rf node_modules package-lock.json
> npm i
> ```

This made no difference, I still see the error :(

**Comment by charbelnicolas** [comment]
Oops, sorry, my bad, it was mongoose's fault. It is working as expected now!

**Comment by alvaropsouza** [comment]
> > Fixed in latest version
> 
> I can still see the warning with version 6.9.10, is there anything else I need to do to get rid of the warning?
> 
> ![ss_2024_02_22_17_44_09](https://private-user-images.githubusercontent.com/14960414/307176528-596a990e-3760-49fe-9711-cd5340ecfeb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzODAyMjQsIm5iZiI6MTcwOTM3OTkyNCwicGF0aCI6Ii8xNDk2MDQxNC8zMDcxNzY1MjgtNTk2YTk5MGUtMzc2MC00OWZlLTk3MTEtY2Q1MzQwZWNmZWIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAyVDExNDUyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2NWMwZTcwNGRjYjNlNjM3ZDcyMDIwMzcwMDk2MjZhMDgwNDQ4Y2U3YTZkMzdlZTkxYzljYzhkOWE4NTZhM2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fmj13cNgewzFveWj7I5j56Z8kDmBNQEjGOo57IjdviU)

still see warning here:
![image](https://github.com/nodemailer/nodemailer/assets/64016030/2759ef9e-77fe-42d4-83bd-23059466f80b)

**Comment by andris9** [comment]
Latest nodemailer does not use the punycode module anymore, so the warning is coming from somewhere else.
